### PR TITLE
Partially upgrade packages (stopping short of breaking changes in starlette)

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): 

Package updates (and a few test updates in response to httpx making a breaking change)

## What is the current behavior?

Most of our deps are old, making it difficult for people to introduce new deps and exposing us to security vulnerabilities

Issue Number: N/A

## What is the new behavior?

- Most packages are updated, the exceptions being some which have breaking changes WRT async. I am out of time to tackle those

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

I've pinned the packages which are actually preventing moving forward, hopefully to be addressed as tech debt soon.

### CODE REVIEW

**Review by solomon-negusse — APPROVED** [review]
A couple of packages not explicitly upgraded in Pip (cachetools, jmespath) got major version upgrade in Pipfile.lock. something to keep an eye on. This bug with pipenv really makes controlled upgrades difficult..

File: tests_v2/unit/app/routes/datasets/test_queries.py [hunk 1]
```diff
@@ -36,9 +36,10 @@ async def test_query_dataset_with_api_key(
         f"/dataset/{dataset_name}/{version_name}/query",
         params=params,
         headers=headers,
+        follow_redirects=True,
     )
 
-    print(response.json())
+    # print(response.json())
```
Comment (gtempus) [inline tests_v2/unit/app/routes/datasets/test_queries.py:42]: I see that these `print` statements are commented out, but in other files, you deleted them entirely. Is there a reason that these are important to keep?

**Review by gtempus — COMMENTED** [review]
Great work moving the needle on getting our packages upgraded, @dmannarino! 🚀 
I just had a couple of comments.

File: tests_v2/unit/app/routes/datasets/test_queries.py (hunk 1 above)
Comment (dmannarino) [reply tests_v2/unit/app/routes/datasets/test_queries.py:42]: Hmm, no. I will remove these entirely as well.

File: Dockerfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ FROM tiangolo/uvicorn-gunicorn-fastapi:python3.8-slim
 ARG ENV
 
 RUN apt-get update -y \
-    && apt-get install --no-install-recommends -y gcc libc-dev musl-dev postgresql-client libpq-dev
+    && apt-get install --no-install-recommends -y curl gcc libc-dev musl-dev postgresql-client libpq-dev
```
[the remark this answers is not in the document]
Comment (dmannarino) [reply Dockerfile]: My apologies, this was apparently copied-over from another branch, in which curl seemed not to be in the image anymore (I think I was using another base image). Removed.

**Pull request merged**